launch/soccer.launch.py

```diff
@@ -38,12 +35,15 @@ def generate_launch_description():
     bringup_dir = Path(get_package_share_directory("rj_robocup"))
     launch_dir = bringup_dir / "launch"
 
+    # TODO(Alex): why are there two defaults? here and at bottom of file in
```

Comment:
You can delete this. Apparently you need both, one is for local values (launch configuration), the other exposes it to the user to change at launch time: https://docs.ros.org/en/galactic/Tutorials/Intermediate/Launch/Using-Substitutions.html#substitutions-example-launch-file. 

This answer explains it better than ros2 documentation:
https://answers.ros.org/question/322874/ros2-what-is-different-between-declarelaunchargument-and-launchconfiguration/

So maybe we only need the declarelaunchargument, but I'm pretty sure we need both.

Reply:
Seems like a dumb ROS quirk to me lol